x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_gaps/trial_license_complete_tier/bulk_action_fill_gaps.ts

```diff
@@ -0,0 +1,556 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from 'expect';
+import { DETECTION_ENGINE_RULES_BULK_ACTION } from '@kbn/security-solution-plugin/common/constants';
+import { BulkActionTypeEnum } from '@kbn/security-solution-plugin/common/api/detection_engine/rule_management';
+import moment from 'moment';
+import { getCustomQueryRuleParams, getSimpleRule } from '../../../utils';
+import { createRule, deleteAllRules } from '../../../../../../common/utils/security_solution';
+
+import { FtrProviderContext } from '../../../../../ftr_provider_context';
+import { deleteAllGaps } from '../../../utils/event_log/delete_all_gaps';
+import { GapEvent, generateGapsForRule } from '../../../utils/event_log/generate_gaps_for_rule';
+import { getGapsByRuleId } from '../../../../../../common/utils/security_solution/detections_response/rules/get_gaps_by_rule_id';
+
+export default ({ getService }: FtrProviderContext): void => {
+  const supertest = getService('supertest');
+  const securitySolutionApi = getService('securitySolutionApi');
+  const es = getService('es');
+  const log = getService('log');
+  const esArchiver = getService('esArchiver');
+
+  const postBulkAction = () =>
+    supertest
+      .post(DETECTION_ENGINE_RULES_BULK_ACTION)
+      .set('kbn-xsrf', 'true')
+      .set('elastic-api-version', '2023-10-31');
+
+  describe('@ess @serverless @skipInServerless Bulk fill rule execution gaps', () => {
+    beforeEach(async () => {
+      await deleteAllRules(supertest, log);
+      await esArchiver.load('x-pack/platform/test/fixtures/es_archives/auditbeat/hosts');
```

Comment:
@rylnd pinging you as a reviewer.

`esArchiver` is proven to cause flakiness. The best thing to do is getting rid of it. On top of that It feels like it's not required for these tests. Anyway, ideally it should be replaced with transparent test setup where it's clear what data is in the entities before running the test logic. For example we [explicitly set up prebuilt rules] (https://github.com/elastic/kibana/blob/main/x-pack/test/security_solution_api_integration/test_suites/detections_response/rules_management/prebuilt_rules/common/import_export/export_prebuilt_rules.ts#L31-L42) with meaningful fields defined.

Obviously it might be challenging for some DE tests. So you may think of planning this work and keep an eye on these tests.